lib/internal/bootstrap/switches/is_main_thread.js

```diff
@@ -6,8 +6,10 @@ const rawMethods = internalBinding('process_methods');
 // TODO(joyeecheung): deprecate and remove these underscore methods
 process._debugProcess = rawMethods._debugProcess;
 process._debugEnd = rawMethods._debugEnd;
-process._startProfilerIdleNotifier = rawMethods._startProfilerIdleNotifier;
-process._stopProfilerIdleNotifier = rawMethods._stopProfilerIdleNotifier;
+
+// Stubs for legacy methods.
+process._startProfilerIdleNotifier = () => {};
+process._stopProfilerIdleNotifier = () => {};
```

Comment:
Just turning them into stubs does not seem an ideal way to handle this.

It would be very surprising for me to see that a function would just silently not behave as it originally did (if I would use the function).

I suggest to at least add a note with `--pending-deprecation` and make this function non-enumerable. Ideally, we could just run CITGM and https://github.com/nodejs/Gzemnid to see how many libraries actually use these. Those should ideally be fixed by us and in that case I would also add a runtime deprecation.

Reply:
@BridgeAR … this PR turns the functionality always-on rather than using these methods to enable or disable it. This isn’t even a breaking change, and in fact adding a runtime deprecation (`--pending-deprecation` or not) would be *far* more invasive, so I can’t really agree here at all.

(That being said, I’ll probably not update the PR until I have looked into Ben’s concerns anyway.)

Reply:
If a user wanted to disable it, it would be a changed behavior?

If we want to prevent usage of an API we should definitely let the user know about it. Just silently doing nothing seems pretty bad to me.

A deprecation notice is IMO also not be invasive. If a user does not want to get notified, it's possible to opt-out of those.

Reply:
> If a user wanted to disable it, it would be a changed behavior?

Only in terms of diagnostic output, not program behavior. Practically speaking, this is not something that’s used in end-user code but in diagnostic platforms.

> If we want to prevent usage of an API we should definitely let the user know about it. Just silently doing nothing seems pretty bad to me.

I don’t want to prevent usage of this API, hence the lack of a deprecation.

> A deprecation notice is IMO also not be invasive. If a user does not want to get notified, it's possible to opt-out of those.

I mean, you know I disagree strongly here.